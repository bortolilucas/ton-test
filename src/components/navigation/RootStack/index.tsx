import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import React from 'react';
import { Colors } from '../../../constants/colors';
import CartProvider from '../../../contexts/CartContext/provider';
import BackButton from '../BackButton';
import CartButton from '../CartButton';
import ProductsListScreen from '../../../screens/products/ProductsListScreen';
import CartScreen from '../../../screens/cart/CartScreen';

export type RootStackParamsList = {
  ProductsList: undefined;
  Cart: undefined;
};

const Stack = createNativeStackNavigator<RootStackParamsList>();

const RootStack = () => {
  return (
    <NavigationContainer
      theme={{
        colors: {
          background: Colors.BACKGROUND,
          text: Colors.TEXT,
          card: Colors.BACKGROUND,
          notification: Colors.TEXT,
          border: Colors.TEXT,
          primary: Colors.TEXT,
        },
        dark: false,
      }}>
      <CartProvider>
        <Stack.Navigator
          screenOptions={{
            headerShadowVisible: false,
            headerBackTitleVisible: false,
            headerTitleAlign: 'center',
            headerLeft: props => <BackButton {...props} />,
            headerRight: () => <CartButton />,
          }}>
          <Stack.Screen
            name="ProductsList"
            component={ProductsListScreen}
            options={{
              title: 'Produtos',
            }}
          />
          <Stack.Screen
            name="Cart"
            component={CartScreen}
            options={{
              title: 'Carrinho',
            }}
          />
        </Stack.Navigator>
      </CartProvider>
    </NavigationContainer>
  );
};

export default RootStack;
