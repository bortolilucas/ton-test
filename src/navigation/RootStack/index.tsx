import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import React from 'react';
import { Colors } from '../../constants/colors';
import CartProvider from '../../contexts/CartContext/provider';
import ProductsList from '../../screens/products/ProductsList';
import CartButton from '../CartButton';

export type RootStackParamsList = {
  ProductsList: undefined;
};

const Stack = createNativeStackNavigator<RootStackParamsList>();

const RootStack = () => {
  return (
    <NavigationContainer
      theme={{
        colors: {
          background: Colors.BACKGROUND,
          text: Colors.TEXT,
          card: Colors.BACKGROUND,
          notification: Colors.TEXT,
          border: Colors.TEXT,
          primary: Colors.TEXT,
        },
        dark: false,
      }}>
      <CartProvider>
        <Stack.Navigator
          screenOptions={{
            headerShadowVisible: false,
          }}>
          <Stack.Screen
            name="ProductsList"
            component={ProductsList}
            options={{
              title: 'Produtos',
              headerRight: () => <CartButton />,
            }}
          />
        </Stack.Navigator>
      </CartProvider>
    </NavigationContainer>
  );
};

export default RootStack;
