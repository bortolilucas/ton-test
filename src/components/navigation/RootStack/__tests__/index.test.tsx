import { render } from '@testing-library/react-native';
import React from 'react';
import RootStack from '..';

describe('RootStack', () => {
  test('should match snapshot', () => {
    const tree = render(<RootStack />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
