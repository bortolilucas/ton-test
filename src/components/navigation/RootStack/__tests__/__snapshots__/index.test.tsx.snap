// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootStack should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=" "
        backgroundColor="#ebebef"
        color="#1c1c1c"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="Produtos"
        titleColor="#1c1c1c"
        topInsetEnabled={false}
        translucent={false}
      >
        <RNSScreenStackHeaderSubview
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="left"
        />
        <RNSScreenStackHeaderSubview
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="right"
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "paddingHorizontal": 15,
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#1c1c1c",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              ÔÅ∫
            </Text>
          </View>
        </RNSScreenStackHeaderSubview>
      </RNSScreenStackHeaderConfig>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "#ebebef",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "top": NaN,
              },
            ]
          }
          testID="loading-indicator"
        >
          <ActivityIndicator
            color="#1c1c1c"
            size="large"
          />
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
